website/src/pages/dictionary/index.js

```diff
@@ -173,8 +224,16 @@ function DataDictionary() {
     );
 
     setFilters({ tiers: [...validDataTiers], attributes: [...validDataAttributes] });
+    setMenuContents(generateMenuContents(schemas));
   }, [dictionary]);
 
+  const filteredSchemas = React.useMemo(() => searchSchemas(), [searchParams]);
+
+  useEffect(() => {
+    setSchemas(filteredSchemas);
+    setMenuContents(generateMenuContents(filteredSchemas));
+  }, [searchParams]);
+
```

Comment:
@hlminh2000 does this look correct to you? I tried different variations but I ended "breaking the rules of hooks" several times.

Reply:
`menuContents` looks like it's also just a transformed version of `schemas` (and maybe a couple other things?) if so, it doesn't need to be a state at all.

This looks correct to me, but the more state there are, the more things we have to sync up and keep track of, which opens up rooms for bug